docs/decisions/0005-monitoring-by-code-owner.rst

```diff
@@ -0,0 +1,48 @@
+Monitoring by Owner
+*******************
+
+Status
+======
+
+Accepted
+
+Context
+=======
+
+It is currently difficult for different teams to have team-based on-calls rotations, alerting and monitoring for various parts of the edx-platform (specifically LMS).
+
+Decision
+========
+
+We will implement a custom metric "code_owner" that can be used in NewRelic (or other monitoring solutions that are made pluggable).
+
+Ultimately, to minimize maintenance, the value of the "owner" metric should be populated using the source-of-truth of ownership of various parts of edx-platform.
+
+See `Rejected Alternatives`_ for details of the decision **not** to split the NewRelic application into multiple NewRelic applications.
+
+Note: "owner" is a MySql reserved word, which NewRelic cautions against using, so we are using "code_owner".
+
+Consequences
+============
+
+This metric should be quickly available for use with custom alerts and custom dashboards.
```

Comment:
Can you explain the expected usage of this metric?  It's unclear to me at this time how valuable this will be for us to maintain manually.

Reply:
1. It won’t be completely manual. I used the spreadsheet to create the mapping. It will be simple to get the mappings from the spreadsheet, but additional work to keep them in sync automatically.
2: We are going to create a quick platform-arch dashboard and platform-arch alerts using this metric, which we could use to start an on-call rotation for our team.

Reply:
hmm.. can we discuss this tomorrow?  I still don't see how this fits into the outcomes we are planning for this quarter and what actually we will gain from this effort.

Reply:
Happy to discuss tomorrow. It is the alternative for the NewRelic app splitting, so if you were on board with that work, maybe you could clarify the value you thought we’d get from that?

Also note that I am nearly ready to deploy this, so we can move on from this as soon as you want and iterate later.

Reply:
Agreed offline that this makes sense.  ADR was updated to hopefully clarify the benefits.

Reply:
@nasthagiri: Hopefully your comments have been addressed in the ADR.  If not, please suggest updates.  Thanks.